### PR TITLE
OCM-8220 | test: Automated case id:52691 User can set availability zones

### PR DESCRIPTION
**Pull request by jameszwang**
zhewang@fedora:~/go/src/rosa$ ginkgo -focus 52691 tests/e2e/
Ginkgo detected a version mismatch between the Ginkgo CLI and the version of Ginkgo imported by your packages:
  Ginkgo CLI Version:
    2.14.0
  Mismatched package versions found:
    2.17.1 used by e2e

  Ginkgo will continue to attempt to run but you may see errors (including flag
  parsing errors) and should either update your go.mod or your version of the
  Ginkgo CLI to match.

  To install the matching version of the CLI run
    go install github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file.  Alternatively you can use
    go run github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file to invoke the matching version of the
  Ginkgo CLI.

  If you are attempting to test multiple packages that each have a different
  version of the Ginkgo library with a single Ginkgo CLI that is currently
  unsupported.
  
Running Suite: e2e tests suite - /home/zhewang/go/src/rosa/tests/e2e
====================================================================
Random Seed: 1716455435

Will run 1 of 80 specs
SSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 1 of 80 Specs in 2741.719 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 79 Skipped
PASS

Ginkgo ran 1 suite in 45m42.676623092s
Test Suite Passed

### CODE REVIEW

File: tests/e2e/test_rosacli_cluster.go [hunk 1]
```diff
@@ -569,3 +569,77 @@ var _ = Describe("Classic cluster creation negavite testing",
...
+				Expect(output.String()).Should(ContainSubstring("--availability-zones"))
+
+				By("Create cluster with availability zones")
+				output, err, _ = clusterService.Create(
+					clusterName, "--region", region, "--availability-zones", availabilityZones,
+				)
+				Expect(err).To(BeNil())
+				Expect(output.String()).To(ContainSubstring(fmt.Sprintf("Cluster '%s' has been created.", clusterName)))
+
+				profilehandler.WaitForClusterReady(rosaClient, clusterName, 60)
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:610]: Please, use profile. We already abandon this way for cluster creation.

**Review by radtriste — COMMENTED** [review]
Shouldn't we check also that Cluster Operators are fine ?

File: tests/e2e/test_rosacli_cluster.go (hunk 1 above)
Comment (jameszwang) [reply tests/e2e/test_rosacli_cluster.go]: fixed. create cluster with profile.

**Comment by jameszwang** [comment]
> Shouldn't we check also that Cluster Operators are fine ?

@radtriste The script followed the test steps of case OCP-52691. Can we merge this PR firstly and create another case/script to check cluster operators?Thanks.

File: tests/e2e/test_rosacli_cluster.go [hunk 2]
```diff
@@ -860,3 +860,80 @@ var _ = Describe("HCP cluster creation negative testing",
...
+		})
+
+		AfterEach(func() {
+			By("Clean remaining resources")
+			rosaClient.CleanResources(clusterID)
+
+		})
+
+		It("User can set availability zones - [id:52691]",
+			labels.Critical, labels.Runtime.Day2,
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:903]: This should be a day1-post

File: tests/e2e/test_rosacli_cluster.go [hunk 3]
```diff
@@ -860,3 +860,80 @@ var _ = Describe("HCP cluster creation negative testing",
...
+					SkipTestOnFeature("create rosa cluster with availability zones")
+				}
+
+				By("List machine pool and check the default one")
+				availabilityZones = profile.ClusterConfig.Zones
+				output, err := machinePoolService.ListMachinePool(clusterID)
+				Expect(err).To(BeNil())
+				mpList, err := machinePoolService.ReflectMachinePoolList(output)
+				Expect(err).To(BeNil())
+				mp := mpList.Machinepool("worker")
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:919]: There is a const defined the default MP name

File: tests/e2e/test_rosacli_cluster.go [hunk 4]
```diff
@@ -860,3 +860,80 @@ var _ = Describe("HCP cluster creation negative testing",
...
+				Expect(err).To(BeNil())
+				Expect(strings.Replace(mp.AvalaiblityZones, " ", "", -1)).To(Equal(availabilityZones))
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:921]: There is function defined in common package to remove the space

File: tests/e2e/test_rosacli_cluster.go (hunk 2 above)
Comment (jameszwang) [reply tests/e2e/test_rosacli_cluster.go]: update to day1-post

File: tests/e2e/test_rosacli_cluster.go (hunk 3 above)
Comment (jameszwang) [reply tests/e2e/test_rosacli_cluster.go]: using const replace "worker"

File: tests/e2e/test_rosacli_cluster.go (hunk 4 above)
Comment (jameszwang) [reply tests/e2e/test_rosacli_cluster.go]: Updated,using common.ReplaceCommaSpaceWithComma() to remove space.

**Pull request merged**